src/Plugin/Field/FieldWidget/OgComplex.php

```diff
@@ -26,13 +30,183 @@
 class OgComplex extends EntityReferenceAutocompleteWidget {
 
   /**
-   * The OG complex widget have a special logic on order to return the groups
-   * that user can reference to.
+   * {@inheritdoc}
    */
   public function formElement(FieldItemListInterface $items, $delta, array $element, array &$form, FormStateInterface $form_state) {
     $parent = parent::formElement($items, $delta, $element, $form, $form_state);
+    // todo: fix the definition in th UI level.
     $parent['target_id']['#selection_handler'] = 'default:og';
+    $parent['target_id']['#selection_settings']['field_mode'] = 'default';
+
     return $parent;
   }
 
+  /**
+   * {@inheritdoc}
+   */
+  public function form(FieldItemListInterface $items, array &$form, FormStateInterface $form_state, $get_delta = NULL) {
+    $parent_form = parent::form($items, $form, $form_state, $get_delta);
+
+    $parent_form['other_groups'] = [];
+
+    // Adding the other groups widget.
+    if ($this->isGroupAdmin()) {
+      $parent_form['other_groups'] = $this->otherGroupsWidget($items, $form_state);
+    }
+
+    return $parent_form;
+  }
+
+  /**
+   * Adding the other groups widget to the form.
+   *
+   * @param $elements
+   *   The widget array.
+   */
+  protected function otherGroupsWidget(FieldItemListInterface $items, FormStateInterface $form_state) {
+    if ($this->fieldDefinition->getTargetEntityTypeId() == 'user') {
+      $description = $this->t('As groups administrator, associate this user with groups you do <em>not</em> belong to.');
+    }
+    else {
+      $description = $this->t('As groups administrator, associate this content with groups you do <em>not</em> belong to.');
+    }
+
+    $field_wrapper = Html::getClass($this->fieldDefinition->getName()) . '-add-another-group';
+
+    $elements = [
+      '#type' => 'container',
+      '#tree' => TRUE,
+      '#title' => $this->t('Other groups'),
+      '#description' => $description,
+      '#prefix' => '<div id="' . $field_wrapper . '">',
+      '#suffix' => '</div>',
+      '#cardinality' => FieldStorageDefinitionInterface::CARDINALITY_UNLIMITED,
+      '#cardinality_multiple' => TRUE,
+      '#theme' => 'field_multiple_value_form',
+      '#field_name' => $this->fieldDefinition->getName(),
+      '#max_delta' => 1,
+    ];
+
+    $elements['add_more'] = [
+      '#type' => 'button',
+      '#value' => $this->t('Add another item'),
+      '#name' => 'add_another_group',
+      '#ajax' => [
+        'callback' => [$this, 'addMoreAjax'],
+        'wrapper' => $field_wrapper,
+        'effect' => 'fade',
+      ],
+    ];
+
+    $start_key = 0;
+
+    $target_type = $this->fieldDefinition->getTargetEntityTypeId();
+    $entity_group_ids = array_map(function($entity) {
```

Comment:
this is holding the groups the entity already belongs to? if so let's add a comment.